### DISCUSSION TITLE
Moved to Lineage from Stock and like it so far but wake on tap working would make it complete!

### DISCUSSION THREAD

**Discussion by saip92** [pgale/lineage_15.1_Installation_TSV#24] [General]
Firstly, thanks for putting together this guide!

While the steps over at XDA are great too, I really wanted to get some native nav bar and status bar since it was driving me crazy but one thing I wish for is wake on tap to work.

I did make some changes like not having the camera shutter as the power button and set the volume buttons to wake the device instead, which is acceptable but waking on tap would be great.

Also, using this recovery on the stock rom is great too since I can still opengapps directly instead of the many steps involved there 😅

**Comment by pgale** [comment]
Tap to wake doesn't work in Lineage 15.1 unfortunately.

**Reply by saip92** [reply]
Ohh... Do you mean in general for LineageOS 15.1? The setting for it does exist.

**Reply by pgale** [reply]
No, for the TSV. The author of the ROM, Deadman mentions it in his release notes. It's a shame as it would be really good to have it working.

**Comment by dwd1** [comment]
@deadman96385 can you fix it please?

**Comment by deadman96385** [comment]
The firmware for the touchscreen has it disabled. So even if you force enable it in the kernel driver it won't work. The only other way would be a hack which leaves the touchscreen on all the time which isn't good.

**Reply by dwd1** [reply]
Its sad :(
How do other android releases/versions solve this problem?
Leave the touchscreen always on is why not good?

And another question.
Is it possible to simply build newer LineageOS versions to this device with device tree (https://github.com/Starfire-development/android_device_lenovo_starfire) and vendor tree (https://github.com/Starfire-development/proprietary_vendor_lenovo_starfire) based on this instructions (https://wiki.lineageos.org/devices/zippo/build/)?

Thank you!

**Reply by deadman96385** [reply]
It is a device specific implementation and not tied to the android release/version. So it depends how each oem does it. Leaving the touchscreen on all the time could damage it in the long term especially for an always powered device like TSV. You can only build the version of lineage that the branch name says. You can use them as a base to make changes to support newer versions though but it isn't simple. Currently there is Lineage 15.1, and initial support for Lineage 19.1 which I believe recovery works but system 
doesn't fully boot yet.

**Reply by saip92** [reply]
The stock image supports it somehow though right? Is there something we can do/contribute?

Sorry if that doesn't make sense. I'm not really well versed with custom rom development. 😅

**Reply by deadman96385** [reply]
As far as I am aware stock does not have it. Where do you see it?

**Reply by pgale** [reply]
@deadman96385 not to hijack this thread but quite a lot of people are having issues with WiFi permanently dropping. Do you have any thoughts on improving/resolving that? Thanks.

**Reply by deadman96385** [reply]
> @deadman96385 not to hijack this thread but quite a lot of people are having issues with WiFi permanently dropping. Do you have any thoughts on improving/resolving that? Thanks. 

I'm not sure how to easily help with it, my 2 units don't have an issue. From the logs I saw early on nothing stood out as different. I've also not been doing as much now that lineage 15.1 works great for using them as a home dashboard with good speakers.

**Reply by saip92** [reply]
> As far as I am aware stock does not have it. Where do you see it?

Well, now that you mention it, I don't see the setting for it. But with the stock, it just works.

To confirm it wasn't something I did, I flashed `CD-18781Y.200628.084` firmware using `edl` and installed lawnchair and navigation bar using `adb` (as found [here](https://xdaforums.com/t/guide-root-unlocking-and-improving-your-lenovo-thinksmart-view-cd-18781y.4584033/)).

I even tried disabling the nav bar just in case that was causing it, but I suppose this version perhaps doesn't turn off the touchscreen?

**Reply by Vartkat** [reply]
Sorry guys, I think I'm missing something. My TSV goes to sleep and then... What ? How do you wake it if double tap doesn't work ? Do we have to reboot each time we want to wake it up ?

**Reply by pgale** [reply]
> Sorry guys, I think I'm missing something. My TSV goes to sleep and then... What ? How do you wake it if double tap doesn't work ? Do we have to reboot each time we want to wake it up ?

Use the side buttons or the camera slider (IIRC) - you can set what those buttons do in the settings.
